### DISCUSSION TITLE
How to custom combobox header template

### DISCUSSION THREAD

**Discussion by skillcrushbinguyen** [microsoft/fluentui-blazor#2712] [General]
Hi FluentUI Community,
I would like to implement a similar UI below in combobox, could you please share some ideas to implement it in FluentUI? Thank you!
![image](https://github.com/user-attachments/assets/f0d8e4f6-b41b-4bf3-9cb2-5465479833ff)

**Comment by dvoituron** [comment]
You should probably use a `FluentTextField` and a `FluentAnchoredRegion`. We've used these components to create the `FluentAutocomplete` component. You can find the implementation [here](https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Components/List/FluentAutocomplete.razor), for inspiration :-)

**Reply by skillcrushbinguyen** [reply]
Unfortunately, inherit from ListComponentBase is impossible because InternalListContext is internal scope :D
Do you have any suggest for me? Thank you!

**Reply by vnbaaij** [reply]
I don't think this is possible to do with `FluentComboBox`. It uses and renders the fluent-combobox web component internally so that will always be used. If you want this component, I'm afraid there is nothing other to do than to build it yourself. You can, of course, use our TextField, Button and Listbox to display the UI but the logic will need to be built. For that you can use the `ListComponentBase` and `InternalListContext` and any other part of our code as inspiration.

If it all works out and you have a nice component, you can perhaps propose to add it to the [fluentui--blazor-community](https://github.com/fluentui-blazor-community) repo.
